salt/proxy/onyx.py

```diff
@@ -0,0 +1,609 @@
+        user = cmd_out[1:-1]
+        configure_terminal_exit()
+        disable()
+        return user
+    except TerminalException as e:
+        log.error(e)
+        return 'Failed to get user'
+
+
+def get_roles(username):
+    '''
+    Get roles that the username is assigned from switch
+
+    .. code-block: bash
+
+        salt '*' onyx.cmd get_roles username=admin
+    '''
+    info = sendline('show user-account {0}'.format(username))
+    roles = re.search(r'^\s*roles:(.*)$', info, re.MULTILINE)
+    if roles:
+        roles = roles.group(1).strip().split(' ')
+    else:
+        roles = []
+    return roles
+
+
+def check_password(username, password, encrypted=False):
+    '''
+    Check if passed password is the one assigned to user
+
+    .. code-block: bash
+
+        salt '*' onyx.cmd check_password username=admin password=admin
+        salt '*' onyx.cmd check_password username=admin \\
+            password='$5$2fWwO2vK$s7.Hr3YltMNHuhywQQ3nfOd.gAPHgs3SOBYYdGT3E.A' \\
+            encrypted=True
+    '''
+    hash_algorithms = {'1': 'md5',
+                       '2a': 'blowfish',
+                       '5': 'sha256',
+                       '6': 'sha512', }
+    user = get_user(username)
+    password_line = user.split('\n')
+    password_line = password_line[1]
+    if not password_line:
+        return None
+    if '!!' in password_line:
+        return False
+    cur_hash = re.search(r'(\$[0-6](?:\$[^$ ]+)+)', str(password_line)).group(0)
+    if encrypted is False:
+        hash_type, cur_salt, hashed_pass = re.search(r'^\$([0-6])\$([^$]+)\$(.*)$', str(cur_hash)).groups()
+        new_hash = gen_hash(crypt_salt=cur_salt, password=password, algorithm=hash_algorithms[hash_type])
+    else:
+        new_hash = password
+    if new_hash == cur_hash:
+        return True
+    return False
+
+
+def check_role(username, role):
+    '''
+    Check if user is assigned a specific role on switch
+
+    .. code-block:: bash
+
+        salt '*' onyx.cmd check_role username=admin role=network-admin
+    '''
+    return role in get_roles(username)
+
+
+def set_password(username, password,
+                 encrypted=False, role=None,
+                 crypt_salt=None, algorithm='sha256'):
+    '''
+    Set users password on switch
+
+    .. code-block:: bash
+
+        salt '*' onyx.cmd set_password admin TestPass
+        salt '*' onyx.cmd set_password admin \\
+            password='$5$2fWwO2vK$s7.Hr3YltMNHuhywQQ3nfOd.gAPHgs3SOBYYdGT3E.A' \\
+            encrypted=True
+    '''
+    password_line = get_user(username)
+    if encrypted is False:
+        if crypt_salt is None:
+            # onyx does not like non alphanumeric characters.  Using the random module from pycrypto
+            # can lead to having non alphanumeric characters in the salt for the hashed password.
+            crypt_salt = secure_password(8, use_random=False)
+        hashed_pass = gen_hash(crypt_salt=crypt_salt, password=password, algorithm=algorithm)
```

Comment:
Same here, regarding an exception being raised on Windows.

Reply:
the same thing for this one as well!